core/src/main/c/share/net.c

```diff
@@ -321,3 +323,34 @@ JNIEXPORT jint JNICALL Java_io_questdb_network_Net_getPeerPort
     }
     return -1;
 }
+
+JNIEXPORT jint JNICALL Java_io_questdb_network_Net_connectAddrInfo
+        (JNIEnv *e, jclass cl, jlong fd, jlong lpAddrInfo) {
+    struct addrinfo *addr = (struct addrinfo *) lpAddrInfo;
+    return connect((int) fd, addr->ai_addr, (int) addr->ai_addrlen);
+}
+
+JNIEXPORT void JNICALL Java_io_questdb_network_Net_freeAddrInfo
+        (JNIEnv *e, jclass cl, jlong address) {
+    if (address != 0) {
+        freeaddrinfo((void *) address);
+    }
+}
+
+JNIEXPORT jlong JNICALL Java_io_questdb_network_Net_getAddrInfo
+        (JNIEnv *e, jclass cl, jlong host, jint port) {
+    struct addrinfo hints;
+    memset(&hints, 0, sizeof(hints));
+    hints.ai_family = AF_UNSPEC;
+    hints.ai_socktype = SOCK_STREAM;
+    struct addrinfo *addr = NULL;
+
+    char _port[32];
+    sprintf(_port, "%d", port);
```

Comment:
Trivial, but 12 would have been enough instead of 32.
Also, `sprintf` is quite a code smell: `snprintf(...)` would be preferable.
It makes no difference here, but if there's a sec review of the code it's not going to stand out.

Reply:
fixed here: https://github.com/questdb/questdb/pull/2239/commits/953e83e079b66a24f0820f40296a89f3fbe44ba4